common/buildcraft/energy/BCEnergyFluids.java

```diff
@@ -110,7 +110,7 @@ private static BCFluid defineFluid(int[] data, int heat, String name) {
             new ResourceLocation(fluidTexture + "_flow"));
         def.setBlockName(name + "_heat_" + heat);
         def.setMapColour(getMapColor(texDark));
-        def.setFlammable(true);
+        def.setFlammable(BCEnergyConfig.enableOilBurn);
```

Comment:
I've only just noticed that the current core makes *every* BC fluid burn, which includes residue. Instead can you fix that as well by adding another element to the data array and declare it at the top of the method, like `sticky`? Perhaps only look at the config value once in this method rather than in the data array.

Reply:
Right sorry. There's a big ugly data array here: 
https://github.com/BuildCraft/BuildCraft/blob/35b4bce36f649421b7ea4a0380db6bcc1f3de543/common/buildcraft/energy/BCEnergyFluids.java#L45 where all of the arguments are taken from